object/slicer/slicer.go

```diff
@@ -79,7 +79,7 @@ type Slicer struct {
 // written to the metadata of all resulting objects as a creation epoch.
 //
 // See also NewSession.
-func New(signer neofscrypto.Signer, cnr cid.ID, owner user.ID, w ObjectWriter, opts Options) *Slicer {
+func New(signer user.Signer, cnr cid.ID, owner user.ID, w ObjectWriter, opts Options) *Slicer {
```

Comment:
Either it obsoletes `owner` or it stays the same (nothing needs `UserID()` inside).

Reply:
`slicer.ObjectWriter` uses `user.Signer`. It's [client implementation](https://github.com/nspcc-dev/neofs-sdk-go/blob/428-simpilfy-user-id-derivation-from-key/client/object_put.go#L311).

It's uses `ObjectPutInit`. But the same like in [ObjectDelete](https://github.com/nspcc-dev/neofs-sdk-go/pull/475#discussion_r1274405118). Compatibility with pool.

Reply:
Yeah, but this makes me wonder if we need an `owner` here at all (`signer.UserID()` is a `user.ID` too). Although non-sticky containers allow putting objects owned by different accounts.

Reply:
I thought the owner and `user.ID` from signer may be different persons and this allows us to set the owner independently from the signer. But if `owner` is an excess info in this case, I will drop it

Reply:
Let's keep it for now.